### PR TITLE
ESSNTL-4081 handle retry for large archives

### PR DESCRIPTION
**Pull request by r14chandra**

### CODE REVIEW

File: src/puptoo/app.py [hunk 1]
```diff
@@ -71,6 +72,19 @@ def handle_signal(signal, frame):
...
 
+def redis_client() :
+    return Redis(host=config.REDIS_HOST, port=config.REDIS_PORT) 
+
+def handle_retries(redis, request_id):
+    count = redis.get(request_id)
+    if not count:
+        count = 0
+    if int(count) == 3:
+        raise Exception("Message process attempts exceeded")
```
Comment (SteveHNH) [inline src/puptoo/app.py:83]: It would be cool to log the request ID here as well. "Message process attempts exceeded for {request_id}"

Comment (r14chandra) [reply src/puptoo/app.py]: Done!

File: src/puptoo/app.py [hunk 2]
```diff
@@ -71,6 +72,19 @@ def handle_signal(signal, frame):
     global running
     running = False
 
+def redis_client() :
```
Comment (csams) [inline src/puptoo/app.py:75]: Use two blank lines between function definitions and surrounding code.

File: src/puptoo/app.py [hunk 3]
```diff
@@ -71,6 +72,19 @@ def handle_signal(signal, frame):
...
+    return Redis(host=config.REDIS_HOST, port=config.REDIS_PORT) 
+
+def handle_retries(redis, request_id):
```
Comment (csams) [inline src/puptoo/app.py:78]: Two blank lines between function definitions and surrounding code.

**Review by csams — APPROVED** [review]
The logic lgtm. My only comments are formatting nits. I'm approving anyway because existing source also has non standard formatting.

File: src/puptoo/app.py (hunk 2 above)
Comment (r14chandra) [reply src/puptoo/app.py:76]: Done!

**Review by SteveHNH — APPROVED** [review]
Looks good to me

File: src/puptoo/utils/config.py [hunk 4]
```diff
@@ -51,6 +51,8 @@ def get_namespace():
     AWS_ACCESS_KEY = os.getenv("STORAGE_ACCESS_KEY", LoadedConfig.objectStore.buckets[0].accessKey)
     AWS_SECRET_KEY = os.getenv("STORAGE_SECRET_KEY", LoadedConfig.objectStore.buckets[0].secretKey)
     USE_SSL = os.getenv("USE_SSL", True)
+    REDIS_HOST = LoadedConfig.inMemoryDb.hostname
+    REDIS_PORT = LoadedConfig.inMemoryDb.port
```
Comment (patilsuraj767) [inline src/puptoo/utils/config.py:55]: @r14chandra I think we also need to get `username` and `password` from clowder config like this - https://github.com/RedHatInsights/ros-backend/blob/main/ros/lib/config.py#L50-L51

Comment (csams) [reply src/puptoo/utils/config.py:55]: @r14chandra had a chance to look back into this?

Comment (r14chandra) [reply src/puptoo/utils/config.py:55]: @csams It would work anyway, right? As I don't see insights-engine has added this config.

Comment (psav) [reply src/puptoo/utils/config.py:55]: I believe it should work without this - an easy way to test would be to run it in the ephemeral environments with a test in the pr_check and see what happens

**Pull request merged**